### PR TITLE
Milestone 3: Interactive form

### PR DESCRIPTION
**Pull request by mbabaliiryn**
## Make form interactive
- Make the BooksForm component Stateful
- Connect the remove book action to the component
- Update readme file

### CODE REVIEW

File: src/containers/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable arrow-body-style */
+/* eslint-disable no-unused-vars */
```
Comment (suretrust) [inline src/containers/BooksList.js:2]: Remove any variable or import that you aren't using, no need to disable the rule

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Book = ({ book, removeBook }) => (
```
Comment (suretrust) [inline src/components/Book.js:5]:
```suggestion
const Book = ({ book, handleBookRemove }) => (
```

File: src/components/Book.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      {book.title}
+      {' '}
+    </td>
+    <td>
+      {' '}
+      {book.category}
+      {' '}
+    </td>
+    <td>
+      <button type="button" onClick={() => removeBook(book.id)}>x</button>
```
Comment (suretrust) [inline src/components/Book.js:23]:
```suggestion
      <button type="button" onClick={handleBookRemove}>x</button>
```

File: src/containers/BooksList.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import Book from '../components/Book';
+import { removeBook } from '../actions';
+
+const BooksList = ({ books, removeBook }) => {
```
Comment (suretrust) [inline src/containers/BooksList.js]:
```suggestion
const BooksList = ({ books, removeBook }) => {
   const handleBookRemove = (book) => removeBook(book)
```

File: src/containers/BooksList.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          <tr>
+            <th>ID</th>
+            <th>Title</th>
+            <th>Category</th>
+            <th>Remove</th>
+          </tr>
+        </thead>
+        <tbody>
+          {books.map(book => (
+            <Book key={book.id} book={book} removeBook={removeBook} />
```
Comment (suretrust) [inline src/containers/BooksList.js:31]:
```suggestion
            <Book key={book.id} book={book} handleBookRemove={handleBookRemove} />
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

* [ ] The `BookList` component should implement `handleBookRemove` which is then passed as props to the `Book` component. [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone3/ImplementsRemoveBook.png)

- [x] Let's fix the linter errors as well.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: src/containers/BooksList.js (hunk 1 above)
Comment (imahnama) [reply src/containers/BooksList.js:2]: noted

File: src/actions/index.js [hunk 6]
```diff
@@ -5,7 +5,7 @@ export const createBook = book => ({
   book,
 });
 
-export const removeBook = id => ({
+export const handleBookRemove = id => ({
```
Comment (suretrust) [inline src/actions/index.js:8]: This should remain `removeBook`.
```suggestion
export const removeBook = id => ({
```

File: src/containers/BooksList.js [hunk 7]
```diff
@@ -1,13 +1,12 @@
...
 import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 
 import Book from '../components/Book';
-import { removeBook } from '../actions';
+import { handleBookRemove } from '../actions';
 
-const BooksList = ({ books, removeBook }) => {
+const BooksList = ({ books, handleBookRemove }) => {
```
Comment (suretrust) [inline src/containers/BooksList.js]: **Implement `handleBookRemove` here.** `BooksList` should accept `removeBook` as props, then inside the component, implement the `handleBookRemove`, which looks similar to the code below. Finally, pass `handleBookRemove` as props to the `Book` component.

```javascript
const handleBookRemove = (book) => removeBook(book)
```

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [x] The suggestion was not about the naming of the variables, please check through the previous code review as well as the new one below to better understand the suggestion.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @mbabaliiryn @imahnama this looks good to me, you have done well so far please merge and move on this is approved :+1:  

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Pull request merged**